File: Boj-CPP/백준 16968번 차량번호판1.cpp
#include<iostream>
using namespace std;
string s;
int solve(string &s1, int idx, char last)
{
	if(s1.length()==idx)
	return 1;
	char start=(s[idx]=='c'?'a':'0');
	char end=(s[idx]=='c'?'z':'9');
	int answer(0);
	for(char i=start; i<=end; i++)
	{
		if(i!=last)
		answer+=solve(s1,idx+1,i);
	}
	return answer;
}
int main()
{
	cin>>s;
	cout<<solve(s,0,',');
}

File: Boj-Kotlin&Java/백준 16968번 차량 번호판1.kt
package 백준

fun main() {
    val br=System.`in`.bufferedReader()
    println("${dfs(br.readLine(),0,' ')}")
}
private fun dfs(s:String,idx:Int,last:Char):Int{
    if(s.length==idx)
        return 1
    val start= if (s[idx]=='c') 'a' else '0'
    val end=if(s[idx]=='c') 'z' else '9'
    var answer=0
    for(i in start..end){
        if(i!=last){
            answer+=dfs(s,idx+1,i)
        }
    }
    return answer
}